### PR TITLE
RHINENG-13940: Konflux related update PostgresQL client to v16 & disable full RHEL …

### PR DESCRIPTION
**Pull request by thearifismail**
…check

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

# Overview

This PR is being created to address [RHINENG-13940](https://issues.redhat.com/browse/RHINENG-13940).
there are a few issues with your dockerfile:
using nonexisting centos 8 repos

    checking for the FULL_RHEL variable (no longer works)
    installing postgresql 13 client instead of a matching version against your DB
    the first 2 are preventing you from moving to konflux,

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by fstavela** [comment]
@thearifismail There is an import error while trying to run the DB upgrade. See the logs from the required PR check.

File: Dockerfile [hunk 1]
```diff
@@ -1,21 +1,15 @@
 FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
 
+ARG pgRepo="https://copr.fedorainfracloud.org/coprs/mmraka/postgresql-16/repo/epel-8/mmraka-postgresql-16-epel-8.repo"
```
[the remark this answers is not in the document]
Comment (MichaelMraka) [reply Dockerfile:3]: better to use  https://copr.fedorainfracloud.org/coprs/g/insights/postgresql-16/repo/epel-8/group_insights-postgresql-16-epel-8.repo

File: Dockerfile [hunk 2]
```diff
@@ -69,6 +63,9 @@ RUN microdnf remove -y $( comm -13 packages-before-devel-install.txt packages-af
     rm packages-before-devel-install.txt packages-after-devel-install.txt && \
     microdnf clean all
 
+# create a symlink to the library missing from postgresql:16.  This may not be needed in future.
+RUN ln -s /usr/lib64/libpq.so.private16-5.16 /usr/lib64/libpq.so.5
```
[the remark this answers is not in the document]
Comment (thearifismail) [reply Dockerfile:67]: The new version of postgresql changed the library named to `libpq.so.private16-5.16`, so setting the symlink works and I don't know who is still using `libpq.so.5` and using sysmlink works and that's why using it.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (thearifismail) [reply Dockerfile]: @kruai udpated the PR with the new repo link provided by @MichaelMraka

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (kruai) [reply Dockerfile:67]: Hmm... Having to add this kind of symlink feels like we're missing a dependency, or one needs to be updated, or something like that. Is it still needed when using the new pgRepo?

[the remark this answers is not in the document]
Comment (thearifismail) [reply Dockerfile]: > Hmm... Having to add this kind of symlink feels like we're missing a dependency, or one needs to be updated, or something like that. Is it still needed when using the new pgRepo?

Good point... running the build without the symlink.

[the remark this answers is not in the document]
Comment (thearifismail) [reply Dockerfile:67]: yes `libpq.so.5` still needed as the library has been renamed to `libpq.so.private16-5.16`.

**Comment by thearifismail** [comment]
NOTE: DO NOT merge this PR until Postgres16 upgrade is complete.

**Pull request merged**